L1Trigger/L1TCaloLayer1/plugins/L1TCaloSummary.cc

```diff
@@ -215,6 +223,32 @@ void L1TCaloSummary<INPUT, OUTPUT>::produce(edm::Event& iEvent, const edm::Event
     //CICADA reads this as a flat vector
     modelInput[14 * i.gctPhi() + (i.gctEta() - 4)] = i.et();
   }
+  // Check if we're using test patterns. If so, we overwrite the inputs with a test pattern
+  if (overwriteWithTestPatterns) {
+    unsigned int evt = iEvent.id().event();
+    unsigned int totalTestPatterns = testPatterns.size();
+    unsigned int patternElement = evt % totalTestPatterns;
+    const edm::ParameterSet& element = testPatterns.at(patternElement);
+    std::stringstream inputStream;
+    std::string PhiRowString;
+
+    edm::LogWarning("L1TCaloSummary") << "Overwriting existing CICADA input with test pattern!\n";
+
+    for (unsigned short int iPhi = 1; iPhi <= 18; ++iPhi) {
```

Comment:
I'm a bit confused, why have this 1-indexed but then subtract the one when you access the array?

Reply:
@antoniovilela @rappoccio Apologies, this fell off my radar as less important over the past few weeks. The number was done this way originally to line it up to the physical phi value as a convenience when I wrote it, but I can clean that up to be a little more appropriate if you prefer.